events_webhook/src/main/java/com/adobe/aio/event/webhook/service/EventVerifier.java

```diff
@@ -0,0 +1,142 @@
+ * governing permissions and limitations under the License.
+ */
+package com.adobe.aio.event.webhook.service;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.adobe.aio.event.webhook.feign.FeignPubKeyService;
+import com.adobe.aio.exception.AIOException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.security.GeneralSecurityException;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.Signature;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EventVerifier {
+
+  private static Logger logger = LoggerFactory.getLogger(EventVerifier.class);
+
+  public static final String ADOBE_IOEVENTS_SECURITY_DOMAIN = "https://static.adobeioevents.com";
+  public static final String ADOBE_IOEVENTS_DIGI_SIGN_1 = "x-adobe-digital-signature-1";
+  public static final String ADOBE_IOEVENTS_DIGI_SIGN_2 = "x-adobe-digital-signature-2";
+  public static final String ADOBE_IOEVENTS_PUB_KEY_1_PATH = "x-adobe-public-key1-path";
+  public static final String ADOBE_IOEVENTS_PUB_KEY_2_PATH = "x-adobe-public-key2-path";
+
+  private final FeignPubKeyService pubKeyService;
+
+
+  EventVerifier(String url) {
+    this.pubKeyService = new FeignPubKeyService(url);
+
+  }
+
+  public EventVerifier() {
+    this(ADOBE_IOEVENTS_SECURITY_DOMAIN);
+  }
+
+  /**
+   * Authenticate the event by checking the target recipient
+   * and verifying the signatures
+   * @param message - the event payload
+   * @param clientId - recipient client id in the payload
+   * @param headers - webhook request headers
+   * @return boolean - TRUE if valid event else FALSE
+   */
+  public boolean authenticateEvent(String message, String clientId, Map<String, String> headers) {
+    if (!isValidTargetRecipient(message, clientId)) {
+      logger.error("target recipient {} is not valid for message {}", clientId, message);
+      return false;
+    }
+    if (!verifyEventSignatures(message, headers)) {
+      logger.error("signatures are not valid for message {}", message);
+      return false;
+    }
+    return true;
+  }
+
+  private boolean verifyEventSignatures(String message, Map<String, String> headers) {
+    String[] digitalSignatures = {headers.get(ADOBE_IOEVENTS_DIGI_SIGN_1),
+        headers.get(ADOBE_IOEVENTS_DIGI_SIGN_2)};
+    String[] pubKeyPaths = {headers.get(ADOBE_IOEVENTS_PUB_KEY_1_PATH),
+        headers.get(ADOBE_IOEVENTS_PUB_KEY_2_PATH)};
+    return verifySignature(message, pubKeyPaths, digitalSignatures);
+  }
+
+  private boolean verifySignature(String message, String[] publicKeyPaths, String[] signatures) {
```

Comment:
I would rather see  `verifySignature(String message, Public publicKey, String signature)` instead here.
wdyt ?
then in `verifyEventSignatures` above 
you would have something like

`verifySignature(message,  getAioEventsPublicKey(headers.get(ADOBE_IOEVENTS_PUB_KEY_1_PATH)), headers.get(ADOBE_IOEVENTS_DIGI_SIGN_1)) ||
verifySignature(message,  getAioEventsPublicKey(headers.get(ADOBE_IOEVENTS_PUB_KEY_2_PATH)), headers.get(ADOBE_IOEVENTS_DIGI_SIGN_2))`

and adding a javadoc explaining the cdn cache and why having onlu one signature is good enough

Reply:
okay. done

Reply:
I see you chose to pass `String` instead of the stronger `PublicKey` type...
we could, but could you also please add the javadoc explaining why having only one signature valid is good enough ?

Reply:
added javadoc

Reply:
well no you did not.
Could you please add it to the javadoc of the public method ?